backend/module/chaos/serverexperimentation/rtds/rtds_cache.go

```diff
@@ -242,86 +257,83 @@ func setSnapshot(snapshotCache cacheWrapper, rtdsLayerName string, cluster strin
-				percentageKey = fmt.Sprintf(HTTPPercentageWithDownstream, ingressPrefix, target.GetDownstreamCluster())
-				faultKey = fmt.Sprintf(HTTPStatusWithDownstream, ingressPrefix, target.GetDownstreamCluster())
+				percentageKey = fmt.Sprintf(HTTPPercentageWithDownstream, ingressPrefix, downstreamCluster)
+				faultKey = fmt.Sprintf(HTTPStatusWithDownstream, ingressPrefix, downstreamCluster)
 			}
 
 		default:
-			logger.Errorw("Invalid fault injection cluster found", "upstream", target.GetUpstreamCluster(), "downstream", target.GetDownstreamCluster(), "faultInjectionCluster", faultInjectionCluster)
+			logger.Errorw("unknown enforcer %v", httpFaultConfig)
 		}
 
-	case *serverexperimentation.TestConfig_Latency:
-		latency := testConfig.GetLatency()
-		percentageValue = latency.Percent
-		faultValue = latency.DurationMs
+	case *serverexperimentation.HTTPFaultConfig_LatencyFault:
+		latency := httpFaultConfig.GetLatencyFault()
+		percentageValue = latency.GetPercentage().GetPercentage()
+		faultValue = latency.GetLatencyDuration().GetFixedDurationMs()
 
-		switch faultInjectionCluster {
-		case serverexperimentation.FaultInjectionCluster_FAULTINJECTIONCLUSTER_DOWNSTREAM:
+		switch interface{}(httpFaultConfig.GetFaultTargeting().GetEnforcer()).(type) {
+		case *serverexperimentation.FaultTargeting_DownstreamEnforcing:
 			// Latency External Fault
-			percentageKey = fmt.Sprintf(LatencyPercentageForExternal, egressPrefix, target.GetUpstreamCluster())
-			faultKey = fmt.Sprintf(LatencyDurationForExternal, egressPrefix, target.GetUpstreamCluster())
+			percentageKey = fmt.Sprintf(LatencyPercentageForExternal, egressPrefix, upstreamCluster)
+			faultKey = fmt.Sprintf(LatencyDurationForExternal, egressPrefix, upstreamCluster)
 
-		case serverexperimentation.FaultInjectionCluster_FAULTINJECTIONCLUSTER_UPSTREAM:
+		case *serverexperimentation.FaultTargeting_UpstreamEnforcing:
 			// Latency Internal Fault for all downstream services
-			if target.GetDownstreamCluster() == "" {
+			if downstreamCluster == "" {
 				percentageKey = fmt.Sprintf(LatencyPercentageWithoutDownstream, ingressPrefix)
 				faultKey = fmt.Sprintf(LatencyDurationWithoutDownstream, ingressPrefix)
 			} else {
 				// Latency Internal Fault for a given downstream services
-				percentageKey = fmt.Sprintf(LatencyPercentageWithDownstream, ingressPrefix, target.GetDownstreamCluster())
-				faultKey = fmt.Sprintf(LatencyDurationWithDownstream, ingressPrefix, target.GetDownstreamCluster())
+				percentageKey = fmt.Sprintf(LatencyPercentageWithDownstream, ingressPrefix, downstreamCluster)
+				faultKey = fmt.Sprintf(LatencyDurationWithDownstream, ingressPrefix, downstreamCluster)
 			}
 
 		default:
-			logger.Errorw("Invalid fault injection cluster found", "upstream", target.GetUpstreamCluster(), "downstream", target.GetDownstreamCluster(), "faultInjectionCluster", faultInjectionCluster)
+			logger.Errorw("unknown enforcer %v", httpFaultConfig)
```

Comment:
wdyt about returning errors on these error conditions so we bail out on generating runtime keys in this case?